File: project-code-main1/project-code-main/app/frontend/src/api/services/logging.ts
import type { components } from '../../../generated/schema';
import { api } from '../api';

export type ErrorLog = components['schemas']['ErrorLog'];

/**
 * Log a message
 * @param error - the error to log
 * @param info - additional info to log
 * @returns void
 */
export const logError = async (error: Error, info?: { componentStack: string }) => {
  const body = {
    component_stack: info?.componentStack,
    error: error.message
  } as ErrorLog;
  const data = await api.post(`/api/logging`, { json: body }).json<void>();
  return data;
};

File: project-code-main1/project-code-main/app/backend/group34_backend/endpoints/logging.py
from flask_app import app
from flask_openapi3 import Tag
from auth.auth import protected
from domain.model import ErrorLog

logging_tag = Tag(name="logging", description="Logging messages")

@app.post('/api/logging', tags=[logging_tag])
def log_message(body: ErrorLog):
    app.logger.error(body.error)
    if body.component_stack:
        app.logger.info(body.component_stack)
    return '', 200